lib/puppet/provider/archive/default.rb

```diff
@@ -96,7 +96,14 @@ def cleanup
   def extract
     if resource[:extract] == :true
       raise(ArgumentError, "missing archive extract_path") unless resource[:extract_path]
-      PuppetX::Bodeco::Archive.new(archive_filepath).extract(resource[:extract_path], nil, resource[:extract_flags])
+      PuppetX::Bodeco::Archive.new(archive_filepath).
+        extract(
+          resource[:extract_path],
+          :custom_command => resource[:extract_command],
```

Comment:
I think this causes a syntax error, you're passing hash key/values as params instead of a hash